test/common/router/config_impl_test.cc

```diff
@@ -315,8 +318,16 @@ TEST_F(RouteMatcherTest, TestRoutes) {
             config.route(genHeaders("bar.baz.com", "/", "GET"), 0)->routeEntry()->clusterName());
   EXPECT_EQ("instant-server",
             config.route(genHeaders(".foo.com", "/", "GET"), 0)->routeEntry()->clusterName());
-  EXPECT_EQ("instant-server",
+  EXPECT_EQ("wildcard",
```

Comment:
Are these tests strong enough to show that the match order follows the specified one (suffix, then prefix wildcards)?

Reply:
@htuch I refactored my tests to handle all cases explicitly.

Something bad is going with the tests from the point I merged with master. I don't think failures result from my code changes, but with no access to test artifacts I can't be sure.